pkg/controllers/apply_controller.go

```diff
@@ -202,6 +202,12 @@ func (r *ApplyWorkReconciler) applyUnstructured(
 		return nil, false, err
 	}
 
+	if !findOwnerReference(curObj.GetOwnerReferences(), workObj.GetOwnerReferences()[0]) {
```

Comment:
this only works if we do not allow any manifest to come with ownerReference. I will think through if that's the right assumption.

Reply:
I'll leave it as it is for now.

Reply:
Please add a TODO that we will block all owner reference in a manifest work

Reply:
Added a TODO.